bftclient/include/bftclient/exception.h

```diff
@@ -0,0 +1,53 @@
+
+// Concord
+//
+// Copyright (c) 2020 VMware, Inc. All Rights Reserved.
+//
+// This product is licensed to you under the Apache 2.0 license (the "License"). You may not use this product except in
+// compliance with the Apache 2.0 License.
+//
+// This product may include a number of subcomponents with separate copyright notices and license terms. Your use of
+// these subcomponents is subject to the terms and conditions of the subcomponent's license, as noted in the LICENSE
+// file.
+
+#pragma once
+
+#include <exception>
+#include <string>
+
+#include "bftclient/config.h"
+
+namespace bft::client {
+
+class BftClientException : public std::exception {
+ public:
+  explicit BftClientException(const std::string& what) : msg(what){};
+
+  virtual const char* what() const noexcept override { return msg.c_str(); }
```

Comment:
Is it intentional to inherit from `std::exception` and overloading `what`?
If we use `std::runtime_error` there is no need for overloading and adding the `msg` member.

Reply:
I just used an example from concord for this. Are you saying to just use a runtime_exception directly instead of a named exception, or inherit from std::runtime_error?

Reply:
Excise me for being ambiguous. I mean that `BftClientException` is in essence a runtime error.
Thus, I propose to inherit from `std:runtime_error`. As a bonus, there is no need to override `what` method.

Reply:
Got it. Thanks! Will fix.